### DISCUSSION TITLE
In resolve inputs keys are optional

### DISCUSSION THREAD

**Discussion by webdeb** [trpc/trpc#1500] [Q&A] [answered]
Hey, great lib just trying it out. Maybe somebody can explain whats wrong with this:

<img width="1144" alt="Bildschirmfoto 2022-02-09 um 16 00 07" src="https://user-images.githubusercontent.com/14992140/153228284-080e77a4-ac20-4209-906a-622bc0c53082.png">

The `getCleanData` function expects values to be strict. But in resolve, the input values are optional typed. How can I mark them as strictly set?

**Comment by KATT** [comment, marked as answer]
https://github.com/colinhacks/zod#installation

> ⚠️ IMPORTANT: You must enable `strict` mode in your `tsconfig.json`. This is a best practice for all TypeScript projects.


Let me know if it doesn't work!

**Reply by webdeb** [reply]
It works! Thank you!

**Comment by icflorescu** [comment]
If `strict` is too much, enabling `strictNullChecks` might also work in some cases.
